apps/modeler/src/cloud_mesh.cpp

```diff
@@ -115,22 +115,20 @@ pcl::modeler::CloudMesh::save(const std::vector<const CloudMesh*>& cloud_meshes,
     return (cloud_meshes[0]->save(filename));
 
   CloudMesh cloud_mesh;
-  for (size_t i = 0, i_end = cloud_meshes.size(); i < i_end; ++ i)
+  for (auto mesh : cloud_meshes)
```

Comment:
No reference here? Is it because there was a copy being created in line 126 (old). If that's the case we're still copying more than we need to. These are the attributes inside `CloudMesh`
```cpp
      PointCloudPtr               cloud_;
      std::vector<pcl::Vertices>  polygons_;

      vtkSmartPointer<vtkPoints>            vtk_points_;
      vtkSmartPointer<vtkCellArray>         vtk_polygons_;
```
and we only need copied `polygons_`.

Reply:
Mmm the current code does not modify `mesh`, or? So we canr iterate with `const auto&`.

Reply:
`mesh` was a pointer, so it was not necessary to use a reference use. I will update it because of latest style guide discussion.

Reply:
Just want to mention that it may be irritating with new style guide now:
```
  for (const auto &mesh : cloud_meshes)
  {
      for (auto &polygon : mesh->polygons_)
      {
....
```
At first loop you think content of mesh is `const`, but it is only the pointer, during the object can be modified.

Reply:
I think if you are writing this code, there can be no confusion. You should know that the vector contents are pointers, and you should understand that taking them by `const` reference only saves you from modifying the vector, not the objects pointed.

If you are a curious reader who wants to understand the code, then yes, after reading first line you may think `mesh` objects are immutable. But from the second line it's clear that a) `mesh` is a pointer; b) whatever it points to is modifiable because otherwise compiler won't allow to take `polygon` by non-`const` reference.

So with a little thinking everything should be clear. And anyway, which approach do you think would be less irritating? For example the way it is now (`auto mesh : cloud_meshes`) trapped both me and Sergio to think about useless copies (although there were none)!

Reply:
Well found the compile issue: `polygon` needed to be a copy here and not a reference. I have no really good idea, which is the best solution to have no irritation, so it's okay ;).